src/handlePress.js

```diff
@@ -16,7 +21,7 @@ export default (
   dispatchRedirect?: boolean,
   e: SyntheticEvent
 ) => {
-  let shouldGo = true
+  let shouldGo = url !== '#'
 
   if (onClick) {
     shouldGo = onClick(e) // onClick can return false to prevent dispatch
```

Comment:
this is gonna pose another problem (i.e. `onClick` will overwrite `shouldGo` back to `true`). this should do the trick instead:

`shouldDispatch = shouldDispatch &&  url !== '#'`